### ISSUE TITLE
Multiple format support

### ISSUE THREAD

**Issue by firedev** [Hacker0x01/react-datepicker#448]
Hi,

Is it possible to display and parse dates in multiple formats? For instance output as "Mar 1, 2016" but allow "1/3/16", "01/03/2016" and the like?

Maybe I could set some global moment.js options?

**Comment by firedev** [comment]
Okay after I forked it I see why it's not an easy task since it tries to convert a string to date on every change.

I hacked around it so it parses date only after you press enter, guess it can be improved, but that'll do for now.

**Comment by rafeememon** [comment]
Cool, yeah I don't think this is something we can easily support but I'm glad you worked around it.  If your use case is that you want to support "similar-looking" formats (e.g. 1/3/16 and 01/03/2016) there might be some tricks you can play with strict parsing on [this line](https://github.com/Hacker0x01/react-datepicker/blob/v0.25.0/src/date_input.jsx#L56); see the [moment docs](http://momentjs.com/docs/#/parsing/string-format/) for more info

**Comment by firedev** [comment]
You can see what I am doing here: https://github.com/firedev/react-datepicker/blob/master/src%2Fdate_input.jsx#L45-L66

I parse date after user presses Enter otherwise it would be impossible to edit field.

**Comment by rafeememon** [comment]
Makes sense -- parsing only after pressing enter would be a breaking change for us but please open a separate issue if that's something you think is worth exploring
